includes/api/class-wc-rest-payment-gateways-controller.php

```diff
@@ -0,0 +1,385 @@
+				'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::EDITABLE ),
+			),
+			'schema' => array( $this, 'get_public_item_schema' ),
+		) );
+	}
+
+	/**
+	 * Check whether a given request has permission to view payment gateways.
+	 *
+	 * @param  WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|boolean
+	 */
+	public function get_items_permissions_check( $request ) {
+		if ( ! wc_rest_check_manager_permissions( 'payment_gateways', 'read' ) ) {
+			return new WP_Error( 'woocommerce_rest_cannot_view', __( 'Sorry, you cannot list resources.', 'woocommerce' ), array( 'status' => rest_authorization_required_code() ) );
+		}
+		return true;
+	}
+
+	/**
+	 * Check if a given request has access to read a payment gateway.
+	 *
+	 * @param  WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|boolean
+	 */
+	public function get_item_permissions_check( $request ) {
+		if ( ! wc_rest_check_manager_permissions( 'payment_gateways', 'read' ) ) {
+			return new WP_Error( 'woocommerce_rest_cannot_view', __( 'Sorry, you cannot view this resource.', 'woocommerce' ), array( 'status' => rest_authorization_required_code() ) );
+		}
+		return true;
+	}
+
+	/**
+	 * Check whether a given request has permission to edit payment gateways.
+	 *
+	 * @param  WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|boolean
+	 */
+	public function update_items_permissions_check( $request ) {
+		if ( ! wc_rest_check_manager_permissions( 'payment_gateways', 'edit' ) ) {
+			return new WP_Error( 'woocommerce_rest_cannot_edit', __( 'Sorry, you are not allowed to edit this resource.', 'woocommerce' ), array( 'status' => rest_authorization_required_code() ) );
+		}
+		return true;
+	}
+
+	/**
+	 * Get payment gateways.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|WP_REST_Response
+	 */
+	public function get_items( $request ) {
+		$this->maybe_define_wp_admin();
+		$payment_gateways = WC()->payment_gateways->payment_gateways();
+		$response         = array();
+		foreach ( $payment_gateways as $payment_gateway_id => $payment_gateway ) {
+			$payment_gateway->id = $payment_gateway_id;
+			$gateway             = $this->prepare_item_for_response( $payment_gateway, $request );
+			$gateway             = $this->prepare_response_for_collection( $gateway );
+			$response[]          = $gateway;
+		}
+		return rest_ensure_response( $response );
+	}
+
+	/**
+	 * Get a single payment gateway.
+	 *
+	 * @param WP_REST_Request $request
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function get_item( $request ) {
+		$this->maybe_define_wp_admin();
+		$gateway = $this->get_gateway( $request );
+
+		if ( is_null( $gateway ) ) {
+			return new WP_Error( 'woocommerce_rest_payment_gateway_invalid', __( "Resource doesn't exist.", 'woocommerce' ), array( 'status' => 404 ) );
```

Comment:
I think we should avoid `doesn't` - its not very professional for an API :) `does not`. This may apply to other endpoints.